### PR TITLE
Moved resample out of parsers to its own sub-package

### LINKED ISSUE #138: move resamplers out of parsers?
### Problem

Why are resamplers in the parsers sub-module, they are not parsers?

### Proposed solution

create a tools sub-package, or simply put the resamplers into tools.py?  Let's discuss what makes the most sense.  @Sparks29032 ?

### PR DESCRIPTION
**Pull request by alisnwu**
closes #138 

Screenshot 2024-11-06 at 5 50 22 PM

@sbillinge ready for review

### CODE REVIEW

File: src/diffpy/utils/parsers/__init__.py [hunk 1]
```diff
@@ -17,11 +17,10 @@
 """
 
 from .loaddata import loadData
-from .resample import resample
 from .serialization import deserialize_data, serialize_data
 
 # silence the pyflakes syntax checker
-assert loadData or resample or True
+assert loadData or True
```
Comment (bobleesj) [inline src/diffpy/utils/parsers/__init__.py:23]: I am afraid this isn't the latest cookiecutting standard for handling `__init__.py` files within `src`. Could you please check whether we have to import these functions lilke `from .loaddata import loadData`?

**Comment by bobleesj** [comment]
@alisnwu Thanks for this quick fix. For future PRs, since pytest is running via GitHub CI, we don't have to attach the screenshot but appreciate the idea.

File: src/diffpy/utils/parsers/__init__.py (hunk 1 above)
Comment (alisnwu) [reply src/diffpy/utils/parsers/__init__.py:23]: No we don't, should I delete these then?

**Comment by sbillinge** [comment]
@alisnwu this sseems to have picked up a conflict

**Pull request closed without merging**